code/common/grafana.q

```diff
@@ -65,23 +73,37 @@ diskvals:{c:(count[x]-ticks)+til ticks;get'[.Q.ind[x;c]]};
 memvals:{get'[?[x;enlist(within;`i;count[x]-ticks,0);0b;()]]};
 catchvals:{@[diskvals;x;{[x;y]memvals x}[x]]};
 
+istype:{[targ;char] (char,del)~2#targ};
+isfunc:istype[;"f"]; istab:istype[;"t"];
+
+// builds body of table response in Json adaptor schema
+tabresponse:{[colname;coltype;rqt] .j.j enlist`columns`rows`type!(flip`text`type!(colname;coltype);catchvals rqt;`table)};
+
 // process a table request and return in JSON format
 tbfunc:{[rqt]
-  rqt:value raze rqt[`targets]`target;
+  rqt: raze rqt[`targets]`target;
+  symname:0b;
+  // if f.t.func, drop first 4 chars
+  rqt:0!value $[isfunc[rqt] & istab[2_rqt]; 4_rqt; 
+                isfunc[rqt]; 2_rqt;
+                istab[rqt]; [rqt: `$del vs rqt; if[2<count rqt; symname: rqt 2]; rqt 1];
+                rqt];
   // get column names and associated types to fit format
   colname:cols rqt;
   coltype:types (0!meta rqt)`t;
-  // build body of response in Json adaptor schema
-  :.j.j enlist`columns`rows`type!(flip`text`type!(colname;coltype);catchvals rqt;`table);
+  // search rqt for sym if symname was set
+  if[-11h=type symname; rqt:?[rqt;enlist(=;sym;enlist symname);0b;()]];
```

Comment:
Does this need to be a functional select? Is this the same as:
```
select from rqt where sym=symname;
```

Reply:
It's not quite the same, `sym` is a variable by the looks of it

Reply:
I don't see `sym` defined elsewhere in the function. @kalkis can let us know.

Reply:
@ThomasSmyth 

`sym` is defined at the top of the script in lines 5-6 here:

```// user defined column name of sym column
sym:@[value;`.grafana.sym;`sym];```

It's equal to whatever value the user has set for .grafana.sym, or `sym if none is given. Therefore the functional select is looking for values in the column named sym where they are equal to the symname.

Reply:
See that now, all good then.